src/App.test.js

```diff
@@ -1,8 +1,43 @@
 import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
 import App from './App';
 
-test('renders learn react link', () => {
+test('renders a list of items', () => {
   render(<App />);
-  const linkElement = screen.getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
+  const listItem = screen.getAllByRole('listitem');
+  expect(listItem[0]).toBeInTheDocument();
+});
+
+test('items can be added, deleted, and edited', () => {
+  render(<App />);
+
+  const addButton = screen.getByRole('button', { name: /add/i });
+  const input = screen.getByRole('textbox');
+
+  userEvent.type(input, 'zebra');
+  userEvent.click(addButton);
+
+  const item = screen.getByText(/zebra/i);
+  expect(item).toBeInTheDocument();
+
+  const deleteButton = screen.getByTestId('3');
+  const zebra = screen.getByText(/zebra/i);
+
+  userEvent.click(deleteButton);
+
+  expect(zebra).not.toBeInTheDocument();
+
+  const editButton = screen.getByTestId('kittens');
```

Comment:
Be sure to check out https://testing-library.com/docs/react-testing-library/cheatsheet/ if you didn't see it in lecture. It provides a hierarchy of best getBys and we should avoid using getByTestId